python/cudf/cudf/core/reshape.py

```diff
@@ -35,16 +35,35 @@ def _align_objs(objs, how="outer"):
             raise ValueError("cannot reindex from a duplicate axis")
```

Comment:
In line 29, we can short-circuit the process of verifying if all index objects are equal by using `any`. I have pushed these changes.